specs/features/Inconsistencies.md

```diff
@@ -0,0 +1,126 @@
+
+* Differences in encoding handling (autodetecting UTF-8 or dealing with OS-specific line endings `\r`, `\r\n`, and `\n`)
+* Features that are easy to implement on one platform might be quite difficult to implement on another. This can sometimes lead to compromised implementations that end up shipping without being brought to spec, or sometimes features that just don't get implemented across all platforms (!!)
+  * This can also present a form of schedule risk because if it so happens that the last platform ends up being the hardest, you can end up with more work than calendar
+* Type coercion in Javascript can cause it to be more permissive than other platforms leading to some constructs like `maxlines: "2"` being valid in JS, but invalid in, say, UWP (if JS code isn't carefully written -- we do, of course, author in Typescript, but we do also have to keep these problems in mind)
+* Distinct implementations means that defaults can vary by platform (e.g. whether the `TextBlock` `wrap` property defaults to `true` if not specified)
+* Differing choices made by devs at implementation time can lead to divergent behaviors (e.g. whether an invalid property value is silently dropped vs. emitting a warning vs. failing to parse the card)
+* Supporting libraries and platform affordances available to us can ease or speed up implementation, but can also be the source of inconsistencies (e.g. differences in how a dev constructs .NET's `DateTime`, UWP's `DateTime`, C++'s `std::chrono::time_point`, and Javascript's `Date` -- instantiating each object with, e.g a string can vary in permissivity)
+  * Indeed, we've had a few issues arise from differences between a hand-rolled solution and a pre-rolled solution (e.g. date/time handling (#1915) and support for markdown (#1984))
+
+### Rendering
+
+Rendering presents another avenue by which inconsistencies can sneak into Adaptive Cards. Some of these problems can be pretty tricky to solve.
+
+* Multiple implementations means that "gotcha" scenarios might not always be covered or may be covered inconsistently (e.g. what happens when you combine `maxLines`, markdown lists, and `wrap: true`?)
+* Platform-provided controls can vary widely in their behaviors and support. Even if divergent behavior is caught during implementation, it may be difficult or tricky to solve completely
+* Some 
+
+## Proposals
+
+* Attempt to reduce code duplication:
```

Comment:
We could also give more options to developers, so they can customize their experiences. for example, we can allow custom markdown library to all non-js libraries as js library does already. In that way, when they file an osistency issue, we can cater them to use custom library. markdown is well standardized, it should be easy for them to find one. I think this will work only if js libary has access to the shared markdown library since the shared markdown libary is the least common denominator.